### DISCUSSION TITLE
failed to run user script

### DISCUSSION THREAD

**Discussion by Haqbani** [chhoumann/quickadd#710] [Q&A]
hi all, i'm trying to add the below script, but when i add it to the chose i get " Invalid macro name" and when i ignore it and use the quick add i get "failed to run user script minmial. Error:  "

code used: 

```
module.exports = async params => {
    const metaEditApi = params.app.plugins.plugins["metaedit"].api;
    const activeFile = params.app.workspace.getActiveFile();
    if (!activeFile && activeFile.extension !== "md") {
        new Notice("No valid active file");
        return;
    }
    
    await params.quickAddApi.executeChoice('Add Activity to Daily Note with Active File Linked', {ProjectName: activeFile.basename})
}

```
<img width="1161" alt="macro iusse 2" src="https://github.com/chhoumann/quickadd/assets/92270934/de21a0c0-974a-461e-bc4e-1e33894c79f1">
<img width="1122" alt="macro issue 1" src="https://github.com/chhoumann/quickadd/assets/92270934/734cd568-9218-4e89-bb0e-8d6733a55820">

**Comment by chhoumann** [comment]
Very late reply - sorry about that.

Two separate things were going on here:

**The blank "Error:" when running the script** comes from the script itself. Two lines can crash it:

1. The guard is inverted: `if (!activeFile && activeFile.extension !== "md")` dereferences `activeFile` exactly when it is null, which throws and produces that empty error message. You want `||`.
2. `params.app.plugins.plugins["metaedit"].api` throws immediately if MetaEdit is not installed - and `metaEditApi` is never used in the script anyway, so the line can simply go.

Fixed version:

```js
module.exports = async (params) => {
    const activeFile = params.app.workspace.getActiveFile();
    if (!activeFile || activeFile.extension !== "md") {
        new Notice("No valid active file");
        return;
    }

    await params.quickAddApi.executeChoice(
        "Add Activity to Daily Note with Active File Linked",
        { ProjectName: activeFile.basename },
    );
};
```

For `executeChoice` to work, a choice with that exact name has to exist in your QuickAdd settings; the `ProjectName` you pass is then available inside it as `{{VALUE:ProjectName}}`.

**The "Invalid macro name" message** came from the old macro-creation flow (it fired on an empty or duplicate name). That flow has since been reworked, so on a current QuickAdd version you will not run into it.

API reference: https://quickadd.obsidian.guide/docs/QuickAddAPI
